src/Button/Button.tsx

```diff
@@ -136,11 +137,21 @@ function getHoverBackgroundColor({
 // I was able to get guarantees to work, but only with very cryptic errors. I
 // decided it'd be best, for the time being, to `throw` if we use things
 // incorrectly.
-interface Props
-  extends React.DetailedHTMLProps<
-    React.ButtonHTMLAttributes<HTMLButtonElement>,
-    HTMLButtonElement
-  > {
+interface Props {
+  /**
+   * Lets you customize how you're going to render this component. You can
+   * either one of two things:
+   *
+   * 1. A string representing the type of element you want rendered, or
+   * 2. A render function. The render function will be called with the props
+   *    that need to be passed to the element. You can use this to change the
+   *    element being rendered or to add your own props that aren't included in
+   *    this component.
+   *
+   * @default "button"
+   */
+  as?: string | ((props: any) => ReturnType<React.FC>);
```

Comment:
Another option would be to _not_ pass `props` and instead use `React.cloneElement` to automatically add them.

That has the pro of a user can't forget and a con of it's magic. I think I like the idea of using `React.cloneElement`.

Reply:
Something to consider - https://twitter.com/0xca0a/status/1147099945368993793 (scroll down to see Ryan Florence weigh in)

Tangentially related, I imagine if a user forgets to pass props then it's immediately obvious that they made a mistake (since it doesn't look like a button) and should revisit the docs

Paired with if the consumer app wrapped this in a `<LinkButton>` component for reuse, there shouldn't be much opportunity for this to forgetfulness to happen

Reply:
I'm happy you're leaning in this direction because that's where I landed as well. Trying to magically add the props falls apart when trying to use emotion.